python/triton/compiler/code_generator.py

```diff
@@ -89,6 +89,23 @@ def _check_fn_args(node, fn, args):
                 )
 
 
+def _is_namedtuple(val):
+    return isinstance(val, type) and issubclass(val, tuple) and hasattr(val, "_fields")
+
+
+def _apply_to_tuple_values(value, fn):
+    if _is_namedtuple(type(value)):
+        fields = value._fields
+    elif isinstance(value, language.tuple):
+        fields = value.type.fields
+    else:
+        assert False, f"Unsupported type {type(value)}"
+
+    vals = [fn(v) for v in value]
+    types = [v.type for v in vals]
```

Comment:
While you're at it, it might be worth fixing this to work with tuples containing None (which will decay here and no longer have a type). Locally I have this as `[v.type if v is not None else constexpr for v in vals]` but I'm not entirely sure this is right.

Reply:
I do not think we even support `return None`.

I would defer it to another PR. Seems like there are more problems